server/gitrest/packages/gitrest-base/src/utils/gitWholeSummaryManager.ts

```diff
@@ -101,45 +141,83 @@ function getSummaryObjectFromWholeSummaryTreeEntry(entry: WholeSummaryTreeEntry)
     throw new NetworkError(400, `Unknown entry type: ${entry.type}`);
 }
 
-interface IFullGitTree {
-    tree: ITree;
-    blobs: Record<string, IBlob>;
-}
-function containsFullGitTree(gitTree: ITree): boolean {
-    return gitTree.tree.length === 1
-        && gitTree.tree[0].type === "blob"
-        && gitTree.tree[0].path.endsWith(fullTreePath);
-}
+/**
+ * Package a Git tree into a Full Git Tree object by (optionally) parsing and unpacking
+ * inner full git tree blobs and (optionally) retrieving all referenced blobs from storage.
+ * 
+ * @param gitTree - Git tree object containing tree entries
+ * @param repoManager - Repository manager to use for retrieving referenced blobs
+ * @param parseInnerFullGitTrees - Whether to parse and unpack inner full git tree blobs
+ * @param retrieveBlobs - Whether to retrieve blobs (other than full git trees) from storage
+ * @param depth - internally tracks recursion depth for potential future logging or protection
+ */
 async function buildFullGitTreeFromGitTree(
     gitTree: ITree,
     repoManager: IRepositoryManager,
+    parseInnerFullGitTrees = true,
+    retrieveBlobs = true,
+    depth = 0,
 ): Promise<IFullGitTree> {
+    let parsedFullTreeBlobs = false;
     const blobPs: Promise<IBlob>[] = [];
-    gitTree.tree.forEach((treeEntry) => {
+    const treeEntries: ITreeEntry[] = [];
+    for (const treeEntry of gitTree.tree) {
         if (treeEntry.type === "blob") {
-            blobPs.push(repoManager.getBlob(treeEntry.sha));
+            if (treeEntry.path.endsWith(fullTreePath) && parseInnerFullGitTrees) {
+                parsedFullTreeBlobs = true;
+                const fullTreeBlob = await repoManager.getBlob(treeEntry.sha);
+                const fullTree = JSON.parse(
+                    fullTreeBlob.encoding === "base64"
+                        // Convert base64 to utf-8 for JSON parsing
+                        ? Buffer.from(fullTreeBlob.content, fullTreeBlob.encoding).toString("utf-8")
+                        : fullTreeBlob.content,
+                ) as IFullGitTree;
+                const builtFullGitTree = await buildFullGitTreeFromGitTree(
+                    fullTree.tree,
+                    repoManager,
+                    true, /* parseInnerFullGitTrees */
+                    // All blobs associated with full git tree are included in the full git tree blob, and
+                    // will not exists in storage individually.
+                    false, /* retrieveBlobs */
+                    depth + 1,
+                );
+                const baseTreeEntryPath = treeEntry.path.replace(fullTreePath, "");
+                treeEntries.push(...builtFullGitTree.tree.tree.map((fullTreeEntry => ({
+                    ...fullTreeEntry,
+                    path: `${baseTreeEntryPath}${fullTreeEntry.path}`,
```

Comment:
There is no need for a slash "/" in between `${baseTreeEntryPath}` and `${fullTreeEntry.path}`, right?

Reply:
Correct. If the path is `".app/.fullTree"`, `baseTreeEntryPath` will equal `".app/"`. If path is just `.fullTree`, then `baseTreeEntryPath` will be `""`